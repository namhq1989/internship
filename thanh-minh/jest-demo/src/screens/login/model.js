import { routerRedux } from 'dva/router'
import { RcNotification } from '../../components'
import { MessageConst, AppConst } from '../../configs'
import i18n from '../../configs/i18n'

export default {
  namespace: 'login',
  state: {},
  effects: {
    *init(payload, { put }) {
      const account = localStorage.getItem(AppConst.localStorage.accountLogin)
      if (account) {
        yield put(routerRedux.push('/home'))
      }
    },
    *login({ payload }, { put }) {
      const languages = i18n.language
      if (localStorage.getItem(AppConst.localStorage.accounts)) {
        const accounts = JSON.parse(localStorage.getItem(AppConst.localStorage.accounts))
        const isExistsAccount = accounts.find(acc => (
          acc.email === payload.email && acc.password === payload.password))
        if (!isExistsAccount) {
          console.log(MessageConst[languages]);
          RcNotification(MessageConst[languages].Login.ErrorAccount, AppConst.notification.error)
          return
        } else {
          localStorage.setItem(AppConst.localStorage.accountLogin, JSON.stringify(isExistsAccount))
        }
        RcNotification(MessageConst[languages].Login.LoginSuccess, AppConst.notification.success)
        yield put(routerRedux.push('/home'))
      } else {
        RcNotification(MessageConst[languages].Login.NotExistAccounts, AppConst.notification.error)
      }
    }
  },
  subscriptions: {
    setup({ dispatch, history }) {
      return history.listen(({ pathname }) => {
        if (pathname === '/login') {
          dispatch({ type: 'init' })
        }
      })
    }
  }
}
