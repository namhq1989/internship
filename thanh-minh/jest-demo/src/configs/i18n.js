import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import MessageConst from './message'

i18n.use(LanguageDetector).init({
  // we init with resources
  resources: {
    en: {
      translations: {
	'Account Kit': 'Login by phone',
        'language-o': 'en_US',
        'login-button': 'Login',
        'email-palceholder': 'Enter email',
        'password-placeholder': 'Enter password',
        'language- label': 'Language',
        'Do not have an account': 'Do not have an account?',
        'Sign-up': 'Sign Up',
        'Username-placeholder': 'Enter username',
        'repassword-placeholder': 'Enter confirm password',
        'age-placeholder': 'Enter age',
        'Do have an account': 'Do you have an account?',
        'log-out': 'Log Out',
        'welcome-page': 'Welcome',
        'come-page': ' to HomePage',
        // validator register
        RequireRegister: MessageConst.en.Register.RequireRegister,
        EmailIsNotValid: MessageConst.en.Register.EmailIsNotValid,
        userNameLengthBetweenFrom2To128Characters: MessageConst.en.Register.userNameLengthBetweenFrom2To128Characters,
        PasswordLengthMustAtLeast6Characters: MessageConst.en.Register.PasswordLengthMustAtLeast6Characters,
        Successfully: MessageConst.en.Register.Successfully,
        ComfirmToFirstPassword: MessageConst.en.Register.ComfirmToFirstPassword,
        AgeMinMax: MessageConst.en.Register.AgeMinMax,
        AgeIntegerNumber: MessageConst.en.Register.AgeIntegerNumber,
        EmailExists: MessageConst.en.Register.EmailExists,
        SuccessRegister: MessageConst.en.Register.SuccessRegister,

        // validator login
        RequireEmailAndPassword: MessageConst.en.Login.RequireEmailAndPassword,
        ErrorAccount: MessageConst.en.Login.ErrorAccount,
        LoginSuccess: MessageConst.en.Login.LoginSuccess,
        NotExistAccounts: MessageConst.en.Login.NotExistAccounts
      }
    },
    vi: {
      translations: {
	'Account Kit': 'Đăng nhập bằng số  điện thoại',
        'language-o': 'vi_VN',
        'login-button': 'Đăng nhập',
        'email-palceholder': 'Nhập email',
        'language- label': 'Ngôn ngữ',
        'Do not have an account': 'Bạn chưa có tài khoản?',
        'Sign-up': 'Đăng ký',
        'Username-placeholder': 'Nhập username',
        'password-placeholder': 'Nhập mật khẩu',
        'repassword-placeholder': 'Nhập xác nhận mật khẩu',
        'age-placeholder': 'Nhập tuổi',
        'Do have an account': 'Bạn có tài khoản?',
        'log-out': 'Đăng xuất',
        'welcome-page': 'Chào mừng',
        'come-page': ' đến với trang HomePage',
        // validator registor
        RequireRegister: MessageConst.vi.Register.RequireRegister,
        EmailIsNotValid: MessageConst.vi.Register.EmailIsNotValid,
        userNameLengthBetweenFrom2To128Characters: MessageConst.vi.Register.userNameLengthBetweenFrom2To128Characters,
        PasswordLengthMustAtLeast6Characters: MessageConst.vi.Register.PasswordLengthMustAtLeast6Characters,
        Successfully: MessageConst.vi.Register.Successfully,
        ComfirmToFirstPassword: MessageConst.vi.Register.ComfirmToFirstPassword,
        AgeMinMax: MessageConst.vi.Register.AgeMinMax,
        AgeIntegerNumber: MessageConst.vi.Register.AgeIntegerNumber,
        EmailExists: MessageConst.vi.Register.EmailExists,
        SuccessRegister: MessageConst.vi.Register.SuccessRegister,

        // validator login
        RequireEmailAndPassword: MessageConst.vi.Login.RequireEmailAndPassword,
        ErrorAccount: MessageConst.vi.Login.ErrorAccount,
        LoginSuccess: MessageConst.vi.Login.LoginSuccess,
        NotExistAccounts: MessageConst.vi.Login.NotExistAccounts
      }
    }
  },
  fallbackLng: 'vi',
  // have a common namespace used around the full app
  ns: ['translations'],
  defaultNS: 'translations',

  keySeparator: false, // we use content as keys

  interpolation: {
    escapeValue: false, // not needed for react!!
    formatSeparator: ','
  },

  react: {
    wait: true
  }
});

export default i18n
