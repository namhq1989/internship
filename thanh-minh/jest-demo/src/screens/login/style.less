.wrapper-login{
  background-color:currentColor;
  min-height: 100vh;
}
.container {
  background: white;
  width: 341px;
  height: 450px;
  margin: 0 auto;
  margin-top: 50px;
  box-shadow: 0 11px 20px currentColor; 
}

.form-login {
  width: 240px;
  margin: 0 auto;
  margin-top: 52px;
}
.input {
  border: none;
  border-radius: 0;
  outline: none;
}
.div-login{
  text-align: center;
  margin-bottom: 0px;
  margin-top: -15px;
  span{
    font-size: 14px;
  }
  a {
    font-size: 16px;
    font-weight: bold;
  }
  
}
.button-login{
  width: 242px;
  font-size: 16px;
  font-weight: bold;

}
.sign-with{
  text-align: center;
  margin-bottom: 30px;
  h3{
    font-size: 14px;
    font-family: arial;
    color: blue;
    font-weight: bold;
  }
}

.account-kit{
  text-align: center;
  margin-bottom: 20px;
  margin-top: -16px;
  Button{
    background: #f60;
    color: aliceblue;
    font-weight: bold;
    font-size: 15px;
  }
 
}
.country-flag{
  text-align: center;
  margin-bottom: 20px;
}