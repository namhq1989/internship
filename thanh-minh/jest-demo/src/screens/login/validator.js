import { AppConst } from '../../configs'
import { KeysConst } from '../../configs/locale'

const PASS_MAX_LENGTH = 6
const validateLogin = {
  isValid: false,
  message: ''
}
/**
 *
 * @param {String} email form ipnut email
 * @param {String} password form input password
 * @returns Object contains: { isValid, message }
 */
const loginValidator = (email, password) => {
  if (!email || !password) {
    validateLogin.message = KeysConst.requireEmailAndPassword
  } else if (!AppConst.regex.email.test(email)) {
    validateLogin.message = KeysConst.emailIsNotValid
  } else if (password.length < PASS_MAX_LENGTH) {
    validateLogin.message = KeysConst.passwordLengthMustAtLeast6Characters
  } else {
    validateLogin.isValid = true
  }
  return validateLogin
}
export {
  loginValidator
}
