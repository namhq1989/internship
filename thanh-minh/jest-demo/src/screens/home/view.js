import React from 'react'
import { connect } from 'dva'
import { Row, Col, Button } from 'antd'
import { translate } from 'react-i18next'
import { LanguageChange } from '../../components'
import { KeysConst } from '../../configs/locale'
import './style.less'

export class HomePageView extends React.Component {
  componentDidMount() {
    this.init()
    this.saveUserInfo()
  }

  /**
   * Call to model for Check whether the user account is logged on
   */
  init = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/init',
    })
  }

  /**
   * Call to model for save user account into store
   */
  saveUserInfo = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/userInfoSave',
    })
  }

  /**
   * Call to model for log out
   */
  handleLogOut = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/logout',
    })
  }

  /**
   * Render register view
   */
  render() {
    const { t, i18n, userInfo } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div>
        <Row type="flex" justify="center">
          <Col span={8} className="wrapper">
            <h1>{t(KeysConst.welcomePage)}
              <strong className="user-info">
                {userInfo.name}
              </strong>
              {t(KeysConst.comePage)}
            </h1>
            <Button
              type="primary"
              onClick={this.handleLogOut}
              className="logout-button"
            >{t(KeysConst.logOut)}
            </Button><br /><br />
            <LanguageChange
              t={t}
              changeLanguage={changeLanguage}
            />
          </Col>
        </Row>
      </div>
    )
  }
}
export default connect(
  ({ home }) => (
    { userInfo: home.userInfo }
  )
)(translate('translations')(HomePageView))
