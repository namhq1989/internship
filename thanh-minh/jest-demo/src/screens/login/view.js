import React from 'react'
import { connect } from 'dva'
import { Col, Row, Form, Icon, Button, Input } from 'antd'
import { translate } from 'react-i18next'
import AccountKit from 'react-facebook-account-kit'
import { ImageConst, AppConst } from '../../configs'
import { RcNotification } from '../../components'
import { loginValidator } from './validator'
import './style.less'

const FormItem = Form.Item
export class LoginView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  handleChange = (event) => {
    const newState = {}
    newState[event.target.name] = event.target.value
    this.setState(newState)
  }

  submitLogin = () => {
    const { t, dispatch } = this.props
    const { email, password } = this.state
    const validator = loginValidator(email, password)
    if (validator.status) {
      const user = { email, password }
      dispatch({
        type: 'login/login',
        payload: user
      })
    } else {
      RcNotification(t(validator.message), AppConst.notification.error)
    }
  }

  render() {
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div className="wrapper-login">
        <Row type="flex" justify="center">
          <Col xs={18} sm={18} md={8} className="container">
            <div className="logo">
              <img src={ImageConst.imageLogin} alt="logo" />
            </div>
            <Form className="form-login">
              <FormItem>
                <Input
                  prefix={<Icon type="user" />}
                  id="email"
                  type="email"
                  placeholder={t('email-palceholder')}
                  name="email"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitLogin}
                />
              </FormItem>
              <FormItem>
                <Input
                  prefix={<Icon type="lock" />}
                  id="password"
                  type="password"
                  placeholder={t('password-placeholder')}
                  name="password"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitLogin}
                />
              </FormItem>
              <FormItem className="div-login-button">
                <Button
                  type="primary"
                  htmlType="button"
                  className="button-login"
                  onClick={this.submitLogin}
                >
                  {t('login-button')}
                </Button>
              </FormItem>
            </Form>
            <div className="account-kit">
              <AccountKit
                appId="195477857985857"
                version="v1.0"
                onResponse={resp => console.log(resp)}
                csrf="Demo-Account-Kit"
                language={t('language-o')}
                loginType="EMAIL"
              >
                {p => <Button {...p}>Account Kit</Button>
              }
              </AccountKit>
            </div>
            <div className="div-login">
              <span>{t('Do not have an account')}&nbsp;&nbsp;</span>
              <a href="/#/register" title="register">
                {t('Sign-up')}
              </a>
            </div>
            <div className="country-flag">
              <span>{t('language- label')}:&nbsp;&nbsp;</span>
              <Button id="vi-button" onClick={() => changeLanguage('vi')}>
                <img src={ImageConst.imageIconVietNamese} alt="vn" />
              </Button>  &nbsp;&nbsp;&nbsp;&nbsp;
              <Button id="en-button" onClick={() => changeLanguage('en')} className="en-flae">
                <img src={ImageConst.imageIconEngland} alt="en" />
              </Button>
            </div>
          </Col>
        </Row>
      </div>
    )
  }
}
export default connect()(translate('translations')(LoginView))
