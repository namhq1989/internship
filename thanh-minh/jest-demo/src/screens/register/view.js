import React from 'react'
import { connect } from 'dva'
import { Col, Row, Form, Icon, Input, Button } from 'antd'
import { translate } from 'react-i18next'
import { ImageConst, AppConst } from '../../configs'
import { RcNotification } from '../../components'
import { registerValidator } from './validator'
import './style.less'


const FormItem = Form.Item
class RegisterView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      email: '',
      password: '',
      rePassword: '',
      age: ''
    }
  }

  handleChange = (event) => {
    event.preventDefault()
    const newState = {}
    newState[event.target.name] = event.target.value
    this.setState(newState)
  }

  submitRegister = () => {
    const { email, password, rePassword, name, age } = this.state
    const { t, dispatch } = this.props
    const validator = registerValidator(name, email, password, rePassword, age)
    if (validator.status) {
      const account = { name, email, password, rePassword, age }
      dispatch({
        type: 'register/register',
        payload: account
      })
    } else {
      RcNotification(t(validator.message), AppConst.notification.error)
    }
  }

  render() {
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div className="wrapper-register">
        <Row type="flex" justify="center">
          <Col xs={18} sm={18} md={8} className="container">
            <div className="logo">
              <img src={ImageConst.imageLogin} alt="logo" />
              <h2 className="title-register">{t('Sign-up')}</h2>
            </div>
            <Form className="form-register">
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="user" />}
                  type="text"
                  placeholder={t('Username-placeholder')}
                  name="name"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="mail" />}
                  type="email"
                  placeholder={t('email-palceholder')}
                  name="email"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="lock" />}
                  type="password"
                  placeholder={t('password-placeholder')}
                  name="password"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="lock" />}
                  type="password"
                  placeholder={t('repassword-placeholder')}
                  name="rePassword"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="idcard" />}
                  type="text"
                  placeholder={t('age-placeholder')}
                  name="age"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="div-register-button">
                <Button
                  type="pi18nKeyrimary"
                  htmlType="button"
                  className="button-register"
                  onClick={this.submitRegister}
                >
                  {t('Sign-up')}
                </Button>
              </FormItem>
            </Form>
            <div className="div-login">
              <span>{t('Do have an account')}&nbsp;&nbsp;</span>
              <a href="/#/login" title="login">
                {t('login-button')}
              </a>
            </div>
            <div className="country-flag">
              <span>{t('language- label')}:&nbsp;&nbsp;</span>
              <Button onClick={() => changeLanguage('vi')}>
                <img src={ImageConst.imageIconVietNamese} alt="vn" />
              </Button>  &nbsp;&nbsp;&nbsp;&nbsp;
              <Button onClick={() => changeLanguage('en')} className="en-flae">
                <img src={ImageConst.imageIconEngland} alt="en" />
              </Button>
            </div>
          </Col>
        </Row>
      </div>
    )
  }
}
export default connect()(translate('translations')(RegisterView))
