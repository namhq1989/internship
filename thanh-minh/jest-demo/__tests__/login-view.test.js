
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import i18n from '../src/configs/i18n'
import { LoginView } from '../src/screens/login/view'

const mck = jest.fn()
describe('Login:', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <LoginView t={mck} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('Update state on input change ( Email, Password )', () => {
    const handleChange = jest.fn()
    const wrapper = shallow(
      <LoginView
        t={mck}
        submitLogin={() => null}
        handleChange={handleChange}
        dispatch={mck}
      />
    )
    wrapper.find('#email').simulate('change', {
      target: { name: 'email', value: 'blah@gmail.com' }
    })
    expect(wrapper.state('email')).toEqual('blah@gmail.com')
    wrapper.find('#password').simulate('change', {
      target: { name: 'password', value: 'minh12345' }
    })
    expect(wrapper.state('password')).toEqual('minh12345')
  })

  it('changeLanguage ', () => {
    const changeLanguage = jest.fn()
    const wrapper = shallow(
      <LoginView
        i18n={i18n}
        t={mck}
        submitLogin={() => null}
        handleChange={() => null}
        changeLanguage={changeLanguage}
        dispatch={mck}
      />
    )
    wrapper.find('#vi-button').simulate('click', 'vi')
    expect(i18n.language).toEqual('vi')
    wrapper.find('#en-button').simulate('click', 'en')
    expect(i18n.language).toEqual('en')
  })

  it('submit call', () => {
    const submitLogin = jest.fn()
    const wrapper = shallow(
      <LoginView
        i18n={i18n}
        t={mck}
        submitLogin={submitLogin}
        handleChange={() => null}
        changeLanguage={() => {}}
        dispatch={mck}
      />
    )
    wrapper.find('.button-login').simulate('click')
  })
})
