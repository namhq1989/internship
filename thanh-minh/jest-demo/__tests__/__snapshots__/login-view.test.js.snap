// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login: renders correctly 1`] = `
<div
  className="wrapper-login"
>
  <Row
    gutter={0}
    justify="center"
    type="flex"
  >
    <Col
      className="container"
      md={8}
      sm={18}
      xs={18}
    >
      <div
        className="logo"
      >
        <img
          alt="logo"
          src={null}
        />
      </div>
      <Form
        className="form-login"
        hideRequiredMark={false}
        layout="horizontal"
        onSubmit={[Function]}
        prefixCls="ant-form"
      >
        <FormItem
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            className="input"
            disabled={false}
            id="email"
            name="email"
            onChange={[Function]}
            onPressEnter={[Function]}
            prefix={
              <Icon
                type="user"
              />
            }
            prefixCls="ant-input"
            type="email"
          />
        </FormItem>
        <FormItem
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            className="input"
            disabled={false}
            id="password"
            name="password"
            onChange={[Function]}
            onPressEnter={[Function]}
            prefix={
              <Icon
                type="lock"
              />
            }
            prefixCls="ant-input"
            type="password"
          />
        </FormItem>
        <FormItem
          className="div-login-button"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Button
            className="button-login"
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
            type="primary"
          />
        </FormItem>
      </Form>
      <div
        className="account-kit"
      >
        <e
          appId="195477857985857"
          csrf="Demo-Account-Kit"
          disabled={false}
          language="en_US"
          loginType="EMAIL"
          onResponse={[Function]}
          version="v1.0"
        />
      </div>
      <div
        className="div-login"
      >
        <span>
            
        </span>
        <a
          href="/#/register"
          title="register"
        />
      </div>
      <div
        className="country-flag"
      >
        <span>
          :  
        </span>
        <Button
          ghost={false}
          id="vi-button"
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
        >
          <img
            alt="vn"
            src={null}
          />
        </Button>
              
        <Button
          className="en-flae"
          ghost={false}
          id="en-button"
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
        >
          <img
            alt="en"
            src={null}
          />
        </Button>
      </div>
    </Col>
  </Row>
</div>
`;
